environment/src/main/java/com/sequenceiq/environment/experience/call/retry/ExperienceCallUtil.java

```diff
@@ -0,0 +1,55 @@
+package com.sequenceiq.environment.experience.call.retry;
+
+import static com.sequenceiq.cloudbreak.util.NullUtil.getIfNotNullOtherwise;
+import static com.sequenceiq.cloudbreak.util.NullUtil.throwIfNull;
+import static com.sequenceiq.environment.experience.call.retry.RetryConstants.MAX_ATTEMPTS;
+
+import javax.ws.rs.core.Response;
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import com.sequenceiq.environment.exception.ExperienceOperationFailedException;
+import com.sequenceiq.environment.experience.ResponseReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ExperienceCallUtil {
+
+    private static final Logger LOCAL_LOGGER = LoggerFactory.getLogger(ExperienceCallUtil.class);
+
+    private ExperienceCallUtil() {
+    }
+
+    public static <T> Optional<T> fetchResultWithRetry(ResponseReader responseReader, URI path, Callable<Response> toCall, Class<T> resolveResponse,
+            String messageUponSingleFailure, Logger logger) {
+        throwIfNull(responseReader, () -> new IllegalArgumentException(ResponseReader.class.getName() + " instance should not be null!"));
+        throwIfNull(toCall, () -> new IllegalArgumentException("No " + Callable.class.getSimpleName() + " was given, therefore no action can be done!"));
+        Logger log = getIfNotNullOtherwise(logger, LOCAL_LOGGER);
+        byte retryCount = 0;
```

Comment:
Why custom retry is necessary? 
I assume the same could be accomplished via the project-wide `@Retryable` approaches.